### PR TITLE
MINOR: Closing resources in SaslClientsWithInvalidCredentialsTest

### PR DESCRIPTION
**Pull request by akatona84**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by akatona84** [comment]
Hi @ijuma , I've encountered a test failure here which caused unexpected threads in the following tests `@Before` phase.
This fixes the issue in case failure happens, only that would be reported (and not together with .classMethod ending failures)

Could you take a look on this please?

**Comment by akatona84** [comment]
@gcsaba2, here's my PR regarding the test fix, thx for the tip for quietly closing consumers

**Review by gcsaba2 — APPROVED** [review]

File: core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala [hunk 1]
```diff
@@ -74,48 +75,58 @@ class SaslClientsWithInvalidCredentialsTest extends IntegrationTestHarness with
...
-    verifyAuthenticationException(producer.partitionsFor(topic))
+    try {
+      verifyAuthenticationException(sendOneRecord(producer, maxWaitMs = 10000))
+      verifyAuthenticationException(producer.partitionsFor(topic))
 
-    createClientCredential()
-    verifyWithRetry(sendOneRecord(producer))
+      createClientCredential()
+      verifyWithRetry(sendOneRecord(producer))
+    } finally producer.close()
```
Comment (omkreddy) [inline core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala]: @akatona84 Thanks for the PR.  Normally producer/consumer instances created in the test are closed in [IntegrationTestHarness.tearDown ](https://github.com/confluentinc/ce-kafka/blob/master/core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala#L152). many core tests extends `IntegrationTestHarness` class.
Do we know, which tests were failed?

Comment (akatona84) [reply core/src/test/scala/integration/kafka/api/SaslClientsWithInvalidCredentialsTest.scala]: Oh, right, it was suspicious that I did not see consumer thread on the failed .classMethod cases :) Thanks.
 testConsumerGroupServiceWithAuthenticationSuccess failed and left adminclient threads for later ones:

> java.lang.AssertionError: Found unexpected threads during `@AfterClass`, allThreads=Set(Test worker, metrics-meter-tick-thread-2, main, Signal Dispatcher, Reference Handler, Finalizer, /0:0:0:0:0:0:0:1:55620 to /0:0:0:0:0:0:0:1:33045 workers Thread 2, kafka-admin-client-thread | adminclient-45, /0:0:0:0:0:0:0:1:55620 to /0:0:0:0:0:0:0:1:33045 workers Thread 3, metrics-meter-tick-thread-1), unexpected=Set(kafka-admin-client-thread | adminclient-45)

**Comment by akatona84** [comment]
Thanks, @omkreddy for explaining how the consumers and prods are closed (I've missed that), updated my PR so only closing things where it is necessary.

**Comment by omkreddy** [comment]
ok to test

**Review by omkreddy — APPROVED** [review]
LGTM

**Pull request merged**